src/app/home/home.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { ProductService } from '../core/ProductService';
+import { Subscription, interval } from 'rxjs';
+import { Product } from '../models/Product';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss'],
+})
+export class HomeComponent implements OnInit {
+  allProducts: Product[];
+  productsToDisplay: Product[] = [];
+  productsSubscription: Subscription;
+
+  constructor(private productService: ProductService) {}
+
+  ngOnInit(): void {
+    this.productsSubscription = this.productService.products.subscribe(
+      (result) => {
+        this.allProducts = result;
+        this.displayRandomProducts();
+      }
+    );
+
+    this.productService.fetchProducts();
+
+    const randomProductsInterval = interval(10000);
+    randomProductsInterval.subscribe(() => {
```

Comment:
a co jesli produktow bedzie 15 000 ? Czy to bedzie optymalne rozwiazanie? :D

Reply:
No nie bedzie ale jak pochytac taki design?

Reply:
Zmienione wedlug ustalen